src/queue.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    let _items = []
+    let _length = 0
+
+    this.getItems = function() {
+      return _items
+    }
+    this.setItems = function( newItems ) {
+      _items = newItems
+    }
+
+    this.getLength = function() {
+      return _length
+    }
+    this.incrementLength = function() {
+      _length++
+    }
+    this.decrementLength = function() {
+      _length--
+    }
+  }
+
+  enqueue( item ) {
+    let items = this.getItems()
+    let index = this.getLength()
+    items[ index ] = item
+    this.incrementLength()
+    this.setItems( items )
+  }
+
```

Comment:
I do see why you wanted this functions, but maybe just doing a this._length++ would be simpler : )